azimuth/modules/model_performance/confusion_matrix.py

```diff
@@ -31,13 +32,21 @@ def compute_on_dataset_split(self) -> List[ConfusionMatrixResponse]:  # type: ig
             ds[self.config.columns.label],
         )
         ds_mng = self.get_dataset_split_manager()
-        class_ids = list(range(ds_mng.get_num_classes()))
+        num_classes = ds_mng.get_num_classes()
+        class_ids = list(range(num_classes))
         cf = confusion_matrix(
             y_true=labels,
             y_pred=predictions,
             labels=class_ids,
             normalize="true" if self.mod_options.cf_normalized else None,
         )
+
+        # Put the rejection class last for the confusion matrix
+        rejection_idx = ds_mng.rejection_class_idx
+        max_idx = num_classes - 1
+        if rejection_idx != max_idx:
+            new_order = [x for x in range(0, max_idx + 1) if x != rejection_idx] + [rejection_idx]
```

Comment:
You could avoid the `for` loop by using two ranges. Also, I think doing `max_idx = num_classes - 1` to then do `max_idx + 1` is more confusing than helpful. I would do
```suggestion
        if rejection_idx != num_classes - 1:
            new_order = [*range(0, rejection_idx), *range(rejection_idx + 1, num_classes), rejection_idx]
```

Reply:
Ho! I see we already have a `class_ids` list. We can reuse it to avoid doing new ranges:
```suggestion
        if rejection_idx != num_classes - 1:
            new_order = class_ids[:rejection_idx] + class_ids[rejection_idx + 1:] + [rejection_idx]
```

Reply:
Ok, cool!!

Reply:
Is it more efficient than a for loop? Or more readable?

Reply:
It is more efficient to avoid the loop. It is probably irrelevant as it take virtually no time anyway, but I find that code that looks sub-optimal is confusing. Like "Why do we need a loop? Am I missing something?". So yeah, realistically, it's more for readability.

Reply:
As it was interesting, I wrote this little script to prove it, with an exaggerated `num_classes = 10_000_000`:
```python
import time
num_classes = 10_000_000
class_ids = list(range(num_classes))
rejection_idx = 12345678
times = []

start_time = time.time()
new_order = [x for x in range(0, num_classes) if x != rejection_idx] + [rejection_idx]
times.append(time.time() - start_time)

start_time = time.time()
new_order = [x for x in class_ids if x != rejection_idx] + [rejection_idx]
times.append(time.time() - start_time)

start_time = time.time()
new_order = [*range(0, rejection_idx), *range(rejection_idx + 1, num_classes), rejection_idx]
times.append(time.time() - start_time)

start_time = time.time()
new_order = class_ids[:rejection_idx] + class_ids[rejection_idx + 1 :] + [rejection_idx]
times.append(time.time() - start_time)

print(f"""
{times[0]} seconds for the loop on range(0, num_classes)
{times[1]} seconds for the loop on class_ids
{times[2]} seconds for two ranges
{times[3]} seconds for two slices of class_ids""")
```
I got
```
1.4199810028076172 seconds for the loop on range(0, num_classes)
1.0802741050720215 seconds for the loop on class_ids
0.6247909069061279 seconds for two ranges
0.4216649532318115 seconds for two slices of class_ids
```